src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Rename/FileMoveNotificationListener.cs

```diff
@@ -21,15 +24,15 @@ internal class FileMoveNotificationListener : IFileMoveNotificationListener
 
         private readonly UnconfiguredProject _unconfiguredProject;
         private readonly IUserNotificationServices _userNotificationServices;
-        private readonly IUnconfiguredProjectVsServices _projectVsServices;
         private readonly Workspace _workspace;
         private readonly IProjectThreadingService _threadingService;
         private readonly IVsService<SVsOperationProgress, IVsOperationProgressStatusService> _operationProgressService;
         private readonly IWaitIndicator _waitService;
         private readonly IRoslynServices _roslynServices;
         private readonly IVsService<SVsSettingsPersistenceManager, ISettingsManager> _settingsManagerService;
 
-        private List<(Renamer.RenameDocumentActionSet Set, string FileName)>? _actions;
+        private readonly Dictionary<string, RenameDocumentActionSet> _renameActionSets = new();
```

Comment:
Changed this to a dictionary because the filepaths *must* be unique. Therefore, the key to the RenameActionSet is the filepath. The *full* filepath itself isn't actually used, though. The filename is pulled out of the path for the progress display.

Reply:
Consider renaming the field to `_renameActionSetByFileName` to make the semantics of key/value clearer.

Reply:
Looking further, I think this is actually `_renameActionSetByFilePath`, possibly with a comment that the path is relative to the project? Please document that if so.

Reply:
I've renamed this to your suggested name and added comments indicating that they use full file-paths.